lib/dsl/lexer.js

```diff
@@ -23,8 +23,51 @@ const Lexer = chevrotain.Lexer;
 
 const tokens = {};
 
+const baseNamePattern = 'baseName';
 const namePattern = /[a-zA-Z_][a-zA-Z_\-\d]*/;
-// All the names tokens of the pegjs implementation have been merged into
+// using start anchor because we are matching with this regexp directly in a custom pattern.
+const namePatternWithStartDigit = /^[\da-zA-Z_][a-zA-Z_\-\d]*/;
+
+/**
+ * Accepts an array of array representing character ranges
+ * e.g: [['a', 'd'], ['0', '2']]
+ *
+ * And returns a set of the "expanded" ranges in charCodes values.
+ * e.g: [97, 99, 98 , 100, 48, 49, 50]
+ *   - 'a'.charCodeAt(0) === 97
+ *   - 'a'.charCodeAt(0) === 100
+ */
+function getRanges(charRanges) {
+  let ranges = [];
+  charRanges.forEach((currRange) => {
+    const startCode = currRange[0].charCodeAt(0);
+    // +1 because lodash range  excludes the end.
+    const endCode = currRange[1].charCodeAt(0) + 1;
+
+    ranges = ranges.concat(_.range(startCode, endCode));
+  });
+
+  return _.uniq(ranges);
+}
+
+// Special NAME Token which follows a baseName config value.
+// It allows starting with numbers or even being only digits.
+const NAME_CAN_START_DIGIT = chevrotain.createToken({
```

Comment:
some wording to do, "can start with digit", nothing critical but I get the meaning

Reply:
Will fix if we move ahead with this change.